tests/unit/test_params.py

```diff
@@ -219,3 +219,78 @@ async def regex_val(text: Annotated[str, Parameter(title="a or b", pattern="[a|b
 
         response = client.get("/val_regex?text=c")
         assert response.status_code == HTTP_400_BAD_REQUEST
+
+
+@pytest.fixture(name="optional_no_default_client")
+def optional_no_default_client_fixture() -> Generator[TestClient, None, None]:
+    @get("/optional-no-default")
+    def handle_optional(key: Optional[str]) -> Dict[str, Optional[str]]:
+        return {"key": key}
+
+    @get("/optional-annotated-no-default")
+    def handle_optional_annotated(param: Annotated[Optional[str], Parameter(query="key")]) -> Dict[str, Optional[str]]:
+        return {"key": param}
+
+    with create_test_client(route_handlers=[handle_optional, handle_optional_annotated]) as client:
+        yield client
+
+
+def test_optional_query_parameter_consistency_no_default_queried_without_param(
+    optional_no_default_client: TestClient,
+) -> None:
+    assert optional_no_default_client.get("/optional-no-default", params={}).json() == {"key": None}
```

Comment:
This behavior is a bit undesirable IMO @litestar-org/maintainers - if a query param is `<smth> | None`, and it automatically defaults to `None` if not provided, that could cause problems. E.g.,:

```py
@get("/top-secret-government-cloud-using-litestar-route")
def armageddon_dead_man_switch(do_not_launch_nukes: Literal["thanks"] | None) -> None:
    if not do_not_launch_nukes:
        big_red_button.press()

# intern forgets to add query parameter to the query.
```
Stupid example, but YKWIM..

Reply:
What would the alternative be to default to? the first `smth` value (`True`, `Literal#1`, etc)?

Reply:
So what's the desirable outcome here?  

Should `do_not_launch_nukes` default to `thanks` instead?

or this?

```python
assert optional_no_default_client.get("/optional-annotated-no-default", params={}).json() == {}
```

Reply:
No default at all IMO, it should be a client error.

Reply:
Explicit better than implicit.

If the developer wants default `None`, they can declare it `key: Optional[str] = None`.

Where the type of the query parameter is `non-optional` and without default, e.g.,:

```py
    @get("/not-optional-no-default")
    def handle_not_optional(key: str) -> Dict[str, str]:
        return {"key": key}
```
And the query omits the parameter, that is a client error already. But for some reason we apply this implicit default when the parameter is optional (which I suspect is a hangover from pydantic signature model behavior).